exporter/clickhousetracesexporter/clickhouse_factory.go

```diff
@@ -78,54 +77,64 @@ func ClickHouseNewFactory(migrations string, datasource string, dockerMultiNodeC
 	}
 }
 
+// Round-robin connection-selection strategy.
+func (f *Factory) selectConn() clickhouse.Conn {
+	if len(f.conns) > 1 {
+		f.conns = append(f.conns[1:], f.conns[0])
+	}
+	return f.conns[0]
+}
+
```

Comment:
This is, currently, not thread-safe. I'm not sure whether parallelism is utilized in `signoz-otel-collector`; should I include a mutex against the list of collectors or something like that?